webapp/channels/src/components/user_settings/plugin/utils.tsx

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export function isValidPluginConfiguration(pluginConfiguration: unknown) {
```

Comment:
I'm not sure if this is how you're intending this function to be used, but If you set the return type of this as `pluginConfiguration is PluginConfiguration`, it'll let TS infer the correct type of the value around this so that you can have 
```ts
let obj: any; // out here, obj is an `any`
if (isValidPluginConfiguration(something)) {
    // in here, obj is a `PluginConfiguration`
}
```

Reply:
I also really like seeing these validation functions since that's sorely lacking for a lot of the Redux code around plugins. We might be able to eventually type the plugin state in a way that TS can figure out the types itself, but these are good to have for now at the very least.

Reply:
I changed this so now it is extracting the configuration, instead of just validating it.

Why? To allow for "future compatibility". Let's imagine that tomorrow we add the input type and a plugin starts using it. In this version, no configuration will be shown at all (since "an input setting is not valid"). The input setting is not valid anymore on this version, but the rest of the settings are. So why not just remove the one that doesn't work?

So... therefore the change in the latest commit.

Reply:
Great idea. I hope we're not changing those too often, but at least during the period where we're adding more setting types, this should give us safety to not worry about plugins breaking older versions of the web app.